spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-sender/src/main/java/com/example/WebController.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.example;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.gcp.pubsub.core.PubSubOperations;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.MessagingGateway;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.gcp.pubsub.outbound.PubSubMessageHandler;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+
+/**
+ * @author João André Martins
+ */
+@RestController
+public class WebController {
+
+	@Autowired
+	private PubsubOutboundGateway messagingGateway;
+
+	@Bean
+	@ServiceActivator(inputChannel = "pubsubOutputChannel")
+	public MessageHandler messageSender(PubSubOperations pubsubTemplate) {
```

Comment:
This bean should probably be defined in a `@Configuration/@SpringBootApplication` class, not the controller.